library/std/src/process.rs

```diff
@@ -1542,6 +1544,29 @@ impl fmt::Display for ExitStatus {
     }
 }
 
+#[unstable(feature = "exit_status_error", issue = "84908")]
+impl Try for ExitStatus {
+    type Output = ();
+    type Residual = Result<Infallible, ExitStatusError>;
+
+    fn branch(self) -> ControlFlow<Self::Residual, Self::Output> {
+        self.0.try_branch()
+    }
+
+    fn from_output((): ()) -> Self {
```

Comment:
If I follow the docs on Try correctly, I think this isn't quite right.

`Try::from_output(x).branch() --> ControlFlow::Continue(x)` to me implies that `type Output = ExitStatus` would be correct here.

I think the current implementation is forcing a synthesis of ExitStatus from (), whereas you actually "want" to just pass along the ExitStatus you're taking in `branch`.

Reply:
This is definitely the least straightforward part of this patch, and definitely deserves scrutiny.
I believed though that it does meet the `Try` laws, because `branch() --> ControlFlow::Continue(x) iff self.code() == 0`.

And I think `type Output = ExitStatus` seems weird to me, because `ExitStatusError` is a subset of `ExitStatus` such that the values intersect, so I felt it was weird in the way that a `Result<ExitStatus, ExitStatusError>` would be strange, since it has values which would be valid in both the `Ok`, and the `Err` branches.

Anyhow, that is the justification for why I did it this way at least.

Reply:
I also noticed this in the documentation for the `Residual` associated type:
> This represents the possible values of the Self type which are not represented by the Output type.

https://doc.rust-lang.org/stable/std/ops/trait.Try.html#associatedtype.Residual

Which `type output = ExitStatus` runs afoul of